sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -519,6 +532,7 @@ private Read() {}
         this.schema = schema;
         this.validate = validate;
         this.codec = codec;
+        this.metadata = ImmutableMap.copyOf(metadata);
```

Comment:
It seems like we are duplicating the map copy multiple times. It makes sense that its only done within the withMetadata call.